python/cuml/dask/common/input_utils.py

```diff
@@ -212,14 +212,16 @@ def _extract_partitions(dask_obj, client=None):
 
     client = default_client() if client is None else client
 
+    dask_obj = client.persist(dask_obj)
     # dask.dataframe or dask.array
     if isinstance(dask_obj, dcDataFrame) or \
             isinstance(dask_obj, daskArray):
-        parts = futures_of(client.compute(dask_obj))
+        parts = client.compute(futures_of(dask_obj))
 
     # iterable of dask collections (need to colocate them)
     elif isinstance(dask_obj, Sequence):
-        parts = [futures_of(client.compute(a)) for a in dask_obj]
+        parts = [client.compute(futures_of(d))
+                 for d in dask_obj]
         to_map = zip(*parts)
         parts = client.compute(list(map(delayed, to_map)))
```

Comment:
Maybe this can be discussed in another issue/PR, but what is the goal with these 2 lines? Wondering if there is a more performant way to get at what is desired here 🙂

Reply:
This is for the case of colocation. An `X` and a `y`, divided into n partitions must have their corresponding partitions on the same GPU. So, the input to the function is `(X, y)`, whose futures then we colocate and ask the client to compute on the same GPU as `(X1, y1), (X2, y2)...(Xn, yn)`

Reply:
Thanks for the explanation! That makes more sense 😄

Also saw [its usage]( https://github.com/rapidsai/cuml/blob/3beb853c461a2ac0eb2e8ac8847592d40949fbe5/python/cuml/dask/common/dask_arr_utils.py#L157-L161 ) here, which gave me some more context. @cjnolet, is there an issue tracking the `map_blocks` problem you ran into? 🙂

Reply:
@jakirkham this is completely unrelated to those issues. Specifically, when two dask arrays (a feature matrix and a set of labels) are passed, their corresponding partitions need to be colocated on the same workers.

Reply:
Gotcha ok. Yeah was just trying to see if we could get `map_blocks` to work here. If the goal is to colocate things, I would think `map_blocks` could work, but I could be missing more context.

Reply:
The goal is to colocate arrays or dataframe, and materialize them. 

The huge memory explosion we discussed (on Tuesday?) was the result of a typo in this function, which is what this PR (and #1891) is fixing. CI didn't catch it in the pytests but Divye & myself both caught it separately.

Reply:
Thanks for the additional context. Yeah I think we are ok leaving this as-is. 🙂

Reply:
How is `dask_obj` structured in the `Sequence` case? Does it already look like `[(X1, y1), (X2, y2)...(Xn, yn)]` or is it `[(X1, X2, ..., Xn), (y1, y2, ..., yn)]`?

If the former, maybe we can drop this whole `if...elif...` block.

If the latter, maybe we should handle `zip(*...)` before `persist`. Then call `futures_of` after that. Should be able to get rid of the Array and DataFrame branch then.

Reply:
The Sequence case is given Dask arrays or Dataframes in the form of [X1, X2, ...] and turns it into a zipped sequence of corresponding partitions [(X1_1, X2_1), (X1_2, X2_2)...]